### PR TITLE
DOCSP-45203 Single Field Indexes

### PR DESCRIPTION
**Pull request by lindseymoore**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-42053>

### Staging Links

<li><a href=https://deploy-preview-100--docs-ruby.netlify.app/indexes>indexes</a></li><li><a href=https://deploy-preview-100--docs-ruby.netlify.app/indexes/single-field-index>indexes/single-field-index</a></li><li><a href=https://deploy-preview-100--docs-ruby.netlify.app/index>index</a></li>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

**Comment by lindseymoore** [comment]
Ignore previous commits before mine. Did a merge instead of a rebase!

File: source/indexes/single-field-index.txt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+field index. The following example creates an index in ascending order on the
+``title`` field:
+
+.. literalinclude:: /includes/indexes/single-field.rb
+   :start-after: start-index-single
+   :end-before: end-index-single
+   :language: ruby
+   :copyable:
+
+Verify the Index's Creation
```
Comment (stephmarie17) [inline source/indexes/single-field-index.txt:76]: [s] Can potentially update this to [avoid using an article](https://www.mongodb.com/docs/meta/style-guide/style/titles-and-headings/#style-and-structure) and be more concise:
```suggestion
Verify Index Creation
```

File: source/indexes/single-field-index.txt [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+---------------------------
+
+You can verify that the index was created by iterating through the ``indexes`` object
+to list the indexes in the collection. You should see an index for ``title`` in 
+the list, as shown in the following output:
```
Comment (stephmarie17) [inline source/indexes/single-field-index.txt:81]: [s] If you want to [avoid using](https://www.mongodb.com/docs/meta/style-guide/writing/use-effective-verbs/#use-modal-verbs-accurately) "should", can maybe edit to something like:
```suggestion
to list the indexes in the collection. The output includes an index for ``title`` in the list, as shown in the following example:
```

File: source/includes/indexes/single-field.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Create a new client and connect to the server
+client = Mongo::Client.new(uri, options)
+
+# start-sample-data
+database = client.use('sample_mflix')
+collection = database[:movies]
+# end-sample-data
+
+# start-index-single
+# Create an index on the "title" field
```
Comment (stephmarie17) [inline source/includes/indexes/single-field.rb:18]: [n] I think some of the narrative comments can have the tense updated similar to the examples in [the code comment guide](https://docs.google.com/document/d/1Bnhh1OsGRUuXo_6yVUASGWiSX6sWuqrEFxPP6sLZDHM/edit?tab=t.0#heading=h.xv6glj2lnzjh).
```suggestion
# Creates an index on the "title" field
```

**Review by stephmarie17 — APPROVED** [review]
LGTM with a few suggestions ✅

File: source/indexes/single-field-index.txt (hunk 2 above)
Comment (lindseymoore) [reply source/indexes/single-field-index.txt:81]: There is a bit of uncertainty, depending on whether the the user is able to successfully create the index. Will leaves as is, if that makes sense to you!

File: source/includes/indexes/single-field.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+client = Mongo::Client.new(uri, options)
+
+# start-sample-data
+database = client.use('sample_mflix')
+collection = database[:movies]
+# end-sample-data
+
+# start-index-single
+# Creates an index on the "title" field
+index = collection.indexes.create_one({ title: 1 })
```
Comment (alexbevi) [inline source/includes/indexes/single-field.rb:19]: You're assigning `index` to the result of the index creation command, but then never using the variable.

File: source/includes/indexes/single-field.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  puts "No document found"
+end
+# end-index-single-query
+
+# start-check-single-index
+# Lists all indexes on the collection
+all_indexes = collection.indexes
+all_indexes.each do |index_spec|
+  puts index_spec.to_json
+end
```
Comment (alexbevi) [inline source/includes/indexes/single-field.rb:39]: Could rewrite this as more idiomatic Ruby by doing:

```ruby
 puts collection.indexes.collect(&:to_json)
```

File: source/includes/indexes/single-field.rb (hunk 4 above)
Comment (alexbevi) [reply source/includes/indexes/single-field.rb:19]: If the return value isn't needed just do:

```ruby
collection.indexes.create_one({ title: 1 })
```

File: source/includes/indexes/single-field.rb (hunk 5 above)
Comment (lindseymoore) [reply source/includes/indexes/single-field.rb:39]: Much more concise, thank you!

File: source/includes/indexes/single-field.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+# Finds a document with the title "Sweethearts"
+filter = { title: 'Sweethearts' }
+doc = collection.find(filter).first
+
+if doc
+  puts doc.to_json
+else
+  puts "No document found"
+end
+# end-index-single-query
```
Comment (alexbevi) [inline source/includes/indexes/single-field.rb:32]: More just a general comment - this has nothing to do with "proving" the index was used ... so why include it?

Comment (lindseymoore) [reply source/includes/indexes/single-field.rb:32]: Hi @alexbevi, this section shows users what kind of query the index will impact. It is not supposed to prove the index's creation. Let me know if this doesn't make sense to you.

Comment (alexbevi) [reply source/includes/indexes/single-field.rb:32]: That makes sense, but it's not clear based on the comment that that's the intention (though I guess it's implied).

Maybe a comment of
```ruby
# Finds a document with the title "Sweethearts", which should use the newly created index
```

Comment (lindseymoore) [reply source/includes/indexes/single-field.rb]: @alexbevi Sounds good, I can add the additional context!

**Pull request merged**